### PR TITLE
Updated benchmark code and refactored transformer implementation

### PR DESCRIPTION
**Pull request by williamberman**
This refactors the transformer into modeling_transformer_v2, 

1. add support for fused norm and mlp blocks
2. remove dead code paths both in generate function and inference code
3. Move conditioning dropout into training script and out of model forward function
4. Simplified benchmarking scripts

### CODE REVIEW

File: training/train_muse.py [hunk 1]
```diff
@@ -693,13 +693,30 @@ def prepare_inputs_and_labels(
...
+                )
+
+                empty_embeds_ = empty_embeds.expand(batch_size, -1, -1)
+                encoder_hidden_states = torch.where(
+                    (encoder_hidden_states * mask).bool(), encoder_hidden_states, empty_embeds_
+                )
+
+                empty_clip_embeds_ = empty_clip_embeds.expand(batch_size, -1)
+                cond_embeds = torch.where((clip_embeds * mask.squeeze(-1)).bool(), clip_embeds, empty_clip_embeds_)
+
```
Comment (williamberman) [inline training/train_muse.py:713]: Move conditional dropout into training script. Really it's usually easiest to usually put this in the dataset class

File: muse/modeling_transformer.py [hunk 2]
```diff
@@ -37,6 +38,8 @@
 except ImportError:
     is_xformers_available = False
 
+MaskGiTUViT = MaskGiTUViT_v2
+
```
Comment (williamberman) [inline muse/modeling_transformer.py:43]: Proxy to new class

**Review by patil-suraj — APPROVED** [review]
